cpp/src/Ice/Timer.cpp

```diff
@@ -196,26 +196,7 @@ Timer::run()
                 }
 
                 _wakeUpTime = first.scheduledTime;
-                try
-                {
-                    _monitor.timedWait(first.scheduledTime - now);
-                }
-                catch(const IceUtil::InvalidTimeoutException&)
-                {
-                    IceUtil::Time timeout = (first.scheduledTime - now) / 2;
-                    while(timeout > IceUtil::Time())
-                    {
-                        try
-                        {
-                            _monitor.timedWait(timeout);
-                            break;
-                        }
-                        catch(const IceUtil::InvalidTimeoutException&)
-                        {
-                            timeout = timeout / 2;
-                        }
-                    }
-                }
+                _conditionVariable.wait_for(lock, chrono::microseconds((first.scheduledTime - now).toMicroSeconds()));
```

Comment:
This was added to fix https://forums.zeroc.com/discussion/5291/exceptions-can-escape-from-iceutil-timer-run

Reply:
Here, we want to wait until the scheduledTime. I don't understand why we use wait_for and not wait_until.

Reply:
How do we do the conversion `IceUtil::Time` to `chorono::time_point`? I was planing to replace IceUtil::Time in a follow up PR